core/src/test/scala/integration/kafka/server/DelayedRemoteFetchTest.scala

```diff
@@ -233,16 +234,16 @@ class DelayedRemoteFetchTest {
   private def buildReadResult(error: Errors,
                               highWatermark: Int = 0,
                               leaderLogStartOffset: Int = 0): LogReadResult = {
-    LogReadResult(
-      exception = if (error != Errors.NONE) Some(error.exception) else None,
-      info = new FetchDataInfo(LogOffsetMetadata.UNKNOWN_OFFSET_METADATA, MemoryRecords.EMPTY),
-      divergingEpoch = None,
-      highWatermark = highWatermark,
-      leaderLogStartOffset = leaderLogStartOffset,
-      leaderLogEndOffset = -1L,
-      followerLogStartOffset = -1L,
-      fetchTimeMs = -1L,
-      lastStableOffset = None)
+    new LogReadResult(
+      new FetchDataInfo(LogOffsetMetadata.UNKNOWN_OFFSET_METADATA, MemoryRecords.EMPTY),
+      Optional.empty(),
+      highWatermark,
+      leaderLogStartOffset,
+      -1L,
+      -1L,
+      -1L,
+      OptionalLong.empty(),
+      if (error != Errors.NONE) Optional.of(error.exception).asInstanceOf[Optional[Throwable]] else Optional.empty[Throwable]())
```

Comment:
Will this be better?
```suggestion
      if (error != Errors.NONE) Optional.of[Throwable](error.exception) else Optional.empty[Throwable]())
```

Reply:
Good catch!